testcases/InstallUpstreamKernel.py

```diff
@@ -128,15 +142,25 @@ def is_url(path):
                 self.cv_SYSTEM.goto_state(OpSystemState.OS)
             else:
                 self.console_thread.console_terminate()
+                log.debug("Compile linux kernel")
+                # as we do not want to install kernel and just do kexec boot of
+                # vmlinux and initrd
+                con.run_command("make -j %d -s && make modules && make modules_install" %
+                                onlinecpus, timeout=self.host_cmd_timeout)
                 cmdline = con.run_command("cat /proc/cmdline")[-1]
                 if self.append_kernel_cmdline:
                     cmdline += " %s" % self.append_kernel_cmdline
                 kern_rel_str = con.run_command(
                     "cat %s/include/config/kernel.release" % linux_path)[-1]
-                initrd_file = con.run_command(
-                    "ls -l /boot/initr*-%s*" % kern_rel_str)[-1].split(" ")[-1]
-                kexec_cmdline = "kexec --initrd %s --command-line=\"%s\" /boot/vmlinuz-%s -l" % (
-                    initrd_file, cmdline, kern_rel_str)
+                vmlinux_file = "/boot/vmlinux-%s" % kern_rel_str
+                initrd_file = "/boot/initramfs-%s" % kern_rel_str
+                # build a initramfs file for the new kernel from the modules
+                # installed
+                con.run_command("dracut %s %s -f" % (initrd_file, kern_rel_str))
+                con.run_command(
+                    "cp -rf %s/vmlinux %s" % (linux_path, vmlinux_file))
```

Comment:
Why skip `make install` if we're copying the vmlinux and initramfs to /boot/ anyway? Also, how well does this work on systems with a limited capacity in /boot? It seems like we should be checking for out of space conditions.